rust/src/lib.rs

```diff
@@ -10,11 +10,23 @@
 mod ffiutil;
 mod includes;
 
+#[cxx::bridge(namespace = "rpmostreecxx")]
```

Comment:
Shouldn't the namespace name be e.g. `rpmostreers` to reflect the fact that it's actually calling into Rust code?

Reply:
Ah but see it's bidirectional, cxx supports Rust calling C++ too and I just submitted https://github.com/coreos/rpm-ostree/pull/2413 to demo that.
Concretely anything listed in `extern "C++"` is expected to be in the target namespace too.

That said it could totally use a better name.  

- `ror::` following the `ror_` naming for bindgen?
- `rpmostree` - Unimaginative but obvious
- Some made up name based on...[a bridge](https://en.wikipedia.org/wiki/List_of_bridges) or something?

Reply:
OTOH we can also change the name later; there's not a lot of code using this right now.  It'll just get slightly more expensive to do so as usage increases.

Reply:
I looked at this a bit more.  It's an interesting topic; partly comes down to whether we see the "cxxrs" part as separate from the C/C++ part.  And the core question there is, do we stick everything in the C++ side into the *same* `rpmostree` namespace?  That'd mean that it'd just be e.g. `download_to_fd` and not `rpmostreecxx::download_to_fd()` when called.

This also conceptually would lead towards dropping the C function prefixing, e.g. instead of `rpmostree_syscore_bump_mtime()` it'd be `rpmostree::syscore_bump_mtime()`.   My feeling right now is that'd be way too much churn; the high level goal here is to sprint towards thinning out the C/C++ side into Rust, not to spend time rewriting into "modern C++".  But there will likely be *some* of the latter.

So I think for now let's just keep `rpmostreecxx::` (and not stick any C++ code into a new namespace) because it highlights the C++/Rust-bridged code in our C-in-C++ code which just works differently for now.

Reply:
Right OK, got it. WDYT about `cxxrs::`? Anyway, yeah we can revisit later.

Also, definitely agreed re. not trying to make too much use of namespaces outside of cxxrs.